src/components/templates/product-availability-modal/available-channels-modal-screen.tsx

```diff
@@ -0,0 +1,99 @@
+import { SalesChannel } from "@medusajs/medusa"
+import React, { useState } from "react"
+import { usePagination, useRowSelect, useTable } from "react-table"
+import SalesChannelAvailabilityTable, {
+  SalesChannelTableActions,
+  useAvailableChannelsModalTableColumns,
+} from "./sales-channel-availability-table"
+
+type AvailableChannelsModalScreenProps = {
+  salesChannels: SalesChannel[]
+  setSelectedSalesChannels: (salesChannels: SalesChannel[]) => void
+}
+
+const AvailableChannelsModalScreen: React.FC<AvailableChannelsModalScreenProps> = ({
+  salesChannels,
+  setSelectedSalesChannels,
+}) => {
+  const limit = 15
+  const numPages = Math.ceil(salesChannels?.length / limit)
+
+  const [offset, setOffset] = useState(0)
+  const [selectedRowIds, setSelectedRowIds] = useState<string[]>([])
+  const [query, setQuery] = useState("")
+
+  const [columns] = useAvailableChannelsModalTableColumns()
+
+  const tableState = useTable(
+    {
+      columns,
+      data: filterSalesChannels(salesChannels),
+      manualPagination: true,
+      initialState: {
+        pageIndex: Math.floor(offset / limit),
+        pageSize: limit,
+      },
+      autoResetPage: false,
+      autoResetSelectedRows: false,
+      getRowId: (row) => row.id,
+      pageCount: numPages,
+    },
+    usePagination,
+    useRowSelect
+  )
+
+  function filterSalesChannels(channels: SalesChannel[]) {
+    if (!query) {
```

Comment:
question: Are all the sales channel loaded locally and then paginate? (in this context I mean)

Reply:
Saleschannels are loaded directly from the product, and yes, then paginated in memory

Reply:
Ok, thankfully we dont expect millions of them 😀 ut i was surprised that everything was done locally